packages/turf-boolean-crosses/index.js

```diff
@@ -0,0 +1,153 @@
+ * both source geometries.
+ *
+ * Boolean-Crosses returns t (TRUE) for only multipoint/polygon, multipoint/linestring, linestring/linestring, linestring/polygon, and linestring/multipolygon comparisons.
+ *
+ * @name booleanCrosses
+ * @param {Geometry|Feature<any>} feature1 GeoJSON Feature or Geometry
+ * @param {Geometry|Feature<any>} feature2 GeoJSON Feature or Geometry
+ * @returns {Boolean} true/false
+ * @example
+ * var cross = turf.booleanCrosses(feature1, feature2);
+ */
+
+module.exports = function (feature1, feature2) {
+    var type1 = getGeomType(feature1);
+    var type2 = getGeomType(feature2);
+    var geom1 = getGeom(feature1);
+    var geom2 = getGeom(feature2);
+
+    switch (type1) {
+    case 'MultiPoint':
+        switch (type2) {
+        case 'LineString':
+            return doMultiPointAndLineStringCross(geom1, geom2);
+        case 'Polygon':
+            return doesMultiPointCrossPoly(geom1, geom2);
+        }
+        throw new Error('feature2 ' + type2 + ' geometry not supported');
+    case 'LineString':
+        switch (type2) {
+        case 'LineString':
+            return doLineStringsCross(geom1, geom2);
+        case 'Polygon':
+            return doLineStringAndPolygonCross(geom1, geom2);
+        }
+        throw new Error('feature2 ' + type2 + ' geometry not supported');
+    default:
+        throw new Error('feature1 ' + type1 + ' geometry not supported');
+    }
+};
+
+function doMultiPointAndLineStringCross(multiPoint, lineString) {
+    var foundIntPoint = false;
+    var foundExtPoint = false;
+    var pointLength = multiPoint.coordinates.length;
+    var i = 0;
+    while (i < pointLength && !foundIntPoint && !foundExtPoint) {
+        for (var i2 = 0; i2 < lineString.coordinates.length - 1; i2++) {
+            var incEndVertices = true;
+            if (i2 === 0 || i2 === lineString.coordinates.length - 2) {
+                incEndVertices = false;
+            }
+            if (isPointOnLineSegment(lineString.coordinates[i2], lineString.coordinates[i2 + 1], multiPoint.coordinates[i], incEndVertices)) {
+                foundIntPoint = true;
+            } else {
+                foundExtPoint = true;
+            }
+        }
+        i++;
+    }
+    return foundIntPoint && foundExtPoint;
+}
+
+function doLineStringsCross(lineString1, lineString2) {
+    var doLinesIntersect = lineIntersect(lineString1, lineString2);
+    if (doLinesIntersect.features.length > 0) {
+        for (var i = 0; i < lineString1.coordinates.length - 1; i++) {
+            for (var i2 = 0; i2 < lineString2.coordinates.length - 1; i2++) {
+                var incEndVertices = true;
+                if (i2 === 0 || i2 === lineString2.coordinates.length - 2) {
+                    incEndVertices = false;
+                }
+                if (isPointOnLineSegment(lineString1.coordinates[i], lineString1.coordinates[i + 1], lineString2.coordinates[i2], incEndVertices)) {
```

Comment:
again you are correct @dpmcmlxxvi this was an oversight late last night. I need to steal and adapt the `lineIntersects` algorithm